RadClass/RadClass.py

```diff
@@ -151,6 +167,10 @@ def march(self):
                 ((new_i + self.integration)
                  >= len(self.processor.timestamps))):
             running = False
+        # ends analysis if stop_time would be reached this iteration
+        elif (self.stop_time is not None and
```

Comment:
I wonder if it would make more sense to constrain the `timesteps` list in `queue_file()` based on both start and stop times and then rely only on the above test?

Reply:
I have not tried implementing this because I fear it would upend the way I track indices using `processor.timestamps`. If I change the length and starting point of `processor.timestamps`, I would have to do the same to `processor.livetime` and I would have to keep track of the offset when accessing spectral data (although the offset is basically just `start_time`). Plus, this would all be changes in `DataSet` since it handles data processing.